script/scaffold/templates/config_flow/integration/__init__.py

```diff
@@ -14,6 +14,8 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up NEW_NAME from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
Why not update the comment:
Like:
`# hass.data.setdefault(DOMAIN, {entry.entry_id: MyApi(...)})`

Reply:
I did it on the base of many codes inside integrations, but this way also makes sense. However I think this may be a bit more understandable for new contributors?

Reply:
I suggest we extend the comment to make it more clear what needs to happen here.

1. Create the API instance.
2. Validate the API connection (and authentication if present).
3. Store the API in hass.data.

The order is important as we don't want to store anything in hass.data until we know we're going to continue setting up the config entry, ie that the API instance is valid.

Reply:
I'll update it soon, potentially on Sunday or Monday (I'm currently a bit unavailable) ;)

Reply:
And done, sorry for making it waiting so long.